### ISSUE TITLE
enforce-slice-style: false positive in an expression context

### ISSUE THREAD

**Issue by connorszczepaniak-wk** [mgechev/revive#1543]
**Describe the bug**
`enforce-slice-style` dictates using `var x []int` rather than `x := []int{}` for slice definitions when configured with the `"nil"` argument, but it also reports the usage of `[]int{}` in an expression context where `var x []int` can't be used

**To Reproduce**
1. I updated revive `go install github.com/mgechev/revive@latest`
2. I run it with the following flags & configuration file:

```shell
revive -config=revive.toml ./...
```

```toml
[rule.enforce-slice-style]
Arguments = ["nil"]
```

With the following code:
```go
package main

func main() {
    foo([]int{})
}

func foo(x []int) {}
```

And got this output:
```
main.go:4:6: use nil slice declaration (e.g. var args []type) instead of []type{}
```

**Expected behavior**
I didn't expect this to be a lint finding because this particular rule seems to be all about declarations, not expressions. The error message makes me think the linter is suggesting the following because it mentions `var args []type`

```go
func main() {
    var x []int
    foo(x)
}
```

So I wonder if either:
1. Only statements should be checked
2. Or, maybe the error message could say something about `use nil instead of []type` when findings are in an expression context; using `foo(nil)` here also suppresses the lint.

**Comment by denisvmedia** [comment]
Have you tried this?

```
foo([]int(nil))
```

**Comment by connorszczepaniak-wk** [comment]
I'm sure that works (as does `foo(nil)`) -- if that's the desired behavior, then can the lint message change for findings in an expression context? Maybe instead of

```
use nil slice declaration (e.g. var args []type) instead of []type{}
```

It could say something like

```
use nil slice declaration (e.g. nil) instead of []type{}
```

The former suggestion doesn't really make sense in an expression context.

**Comment by fzlzjerry** [comment]
Is anyone already working on this? I'd like to take it on.

I reproduced the diagnostic on current master (`46b68c2`), for both an empty literal and `make([]int, 0)` passed to a function. I'd keep the rule's detection unchanged and update the nil-style suggestion to a typed nil expression, such as `[]type(nil)`, which also works in initializers and return values. I'll cover the expression cases and keep the other slice styles unchanged.
